app/models/rule.rb

```diff
@@ -41,18 +42,32 @@ class Rule < ApplicationRecord
     joins(:profile_rules).where.not(profile_rules: { profile_id: nil }).distinct
   }
 
-  def self.from_openscap_parser(op_rule, benchmark_id: nil)
-    rule = find_or_initialize_by(ref_id: op_rule.id,
-                                 benchmark_id: benchmark_id)
+  class << self
+    def from_openscap_parser(op_rule, benchmark_id: nil)
+      rule = find_or_initialize_by(ref_id: op_rule.id,
+                                   benchmark_id: benchmark_id)
 
-    rule.op_source = op_rule
+      rule.op_source = op_rule
 
-    rule.assign_attributes(title: op_rule.title,
-                           description: op_rule.description,
-                           rationale: op_rule.rationale,
-                           severity: op_rule.severity)
+      rule.assign_attributes(title: op_rule.title,
+                             description: op_rule.description,
+                             rationale: op_rule.rationale,
+                             severity: op_rule.severity)
 
-    rule
+      rule
+    end
+
+    def search_by_reference(_, operator, value)
+      conditions = sanitize_sql_for_conditions(
+        ["rule_references.label #{operator} (?)", CGI.unescape(value)]
```

Comment:
I've added now here `(?)` and pass the operator (scoped search automatically translates the operator into a SQL operator), so all possible operators work